src/map/entities/charentity.cpp

```diff
@@ -462,6 +462,32 @@ void CCharEntity::resetPetZoningInfo()
     petZoningInfo.jugSpawnTime = 0;
     petZoningInfo.jugDuration  = 0;
 }
+
+bool CCharEntity::shouldPetPersistThroughZoning()
+{
+    PET_TYPE petType;
+    auto     PPetEntity = dynamic_cast<CPetEntity*>(PPet);
+
+    if (PPetEntity == nullptr && !petZoningInfo.respawnPet)
```

Comment:
Should this be an `||`?

Reply:
This check will exit with `false` early if we have no pet *and* there's no stored pet (`respawnPet == true`).

Hypothetically, is `respawnPet` is set then we already did `setPetZoningInf()` which should always be after `shouldPetPersistThroughZoning()` returned true, but I didn't want to assume that was always the case.

Reply:
Gotcha, thanks.